examples/nice_demo.ipynb

```diff
@@ -0,0 +1,780 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Computes spherical expansion coefficients"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The starting point for the construction of NICE features is an expansion of the neighbor density of atom $i$ -- a _representation_ of the atomic structure in the atom-centered neighborhood $A_i$ of structure $A$\n",
+    "$$\n",
+    "\\langle a n l m|A; \\rho_i\\rangle = \\sum_{j\\in A_i} \\delta_{aa_j}\n",
+    "\\int \\mathrm{d}\\mathbf{r} \\langle n|r\\rangle \\langle lm|\\hat{\\mathbf{r}}\\rangle\n",
+    "\\langle \\mathbf{r} | \\mathbf{r}_{ji}; g\\rangle\n",
+    "$$\n",
+    "In this expression, $\\langle \\mathbf{r} | \\mathbf{r}_{ji}; g\\rangle$ is a localized density function (usually a Gaussian or a Dirac $\\delta$) indicating the position of neighbor $j$, $\\langle \\hat{\\mathbf{r}}| lm\\rangle$ is a spherical harmonic, and $\\langle r | n \\rangle$ a radial basis function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import urllib.request\n",
+    "# a collection of distorted allyl alcohol molecules from the ANI-1 dataset \n",
+    "# (see https://github.com/isayev/ANI1_dataset) with energies and forces computed using DFTB+ \n",
+    "# (see https://www.dftbplus.org/)\n",
+    "url = 'https://raw.githubusercontent.com/cosmo-epfl/librascal-example-data/833b4336a7daf471e16993158322b3ea807b9d3f/inputs/molecule_conformers_dftb.xyz'\n",
+    "# Download the file from `url`, save it in a temporary directory and get the\n",
+    "# path to it (e.g. '/tmp/tmpb48zma.txt') in the `structures_fn` variable:\n",
+    "structures_fn, headers = urllib.request.urlretrieve(url)\n",
+    "structures_fn"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Total number of structure to load\n",
+    "N = 100\n",
+    "\n",
+    "# load the structures\n",
+    "frames = read(structures_fn,':{}'.format(N))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# use librascal to compute expansion coefficients\n",
+    "spherical_expansion_hypers = {\n",
+    "    \"interaction_cutoff\": 3,\n",
+    "    \"max_radial\": 6,\n",
+    "    \"max_angular\": 4,\n",
+    "    \"gaussian_sigma_constant\": 0.3,\n",
+    "    \"gaussian_sigma_type\": \"Constant\",\n",
+    "    \"cutoff_smooth_width\": 0.5,\n",
+    "    \"radial_basis\": \"GTO\",\n",
+    "}\n",
+    "\n",
+    "spex = SphericalExpansion(**spherical_expansion_hypers)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "selframe = frames[8];         # frame and l value used for the test\n",
+    "feat_scaling = 1e6            # just a scaling to make coefficients O(1)\n",
```

Comment:
How would you do that? An additional option "scaling"? That would also help making sure it is applied consistently when you define a potential. To be done in a separate PR though IMO

Reply:
I was wondering whether we could determine the required scaling analytically before computing spherical expansion, and then automatically scale everything (maybe inside the radial integral?). Otherwise, we could rescale after computing the expansion by looking at the mean absolute value, although the very large self contribution coefficient will be an issue for that. I'd rather not have the user have to specify the exact scaling value.

Then once we have a way of automatically scaling the spherical expansion, we could either do it by default or include another hyper for it.

It is definitively for a future PR though!